bin/install-theos

```diff
@@ -383,69 +382,58 @@ linux() {
 					;;
 				suse)
-					sudo zypper install -y $(zypper search libz3 | tail -n 1 | cut -d "|" -f2) zstd
-					# libz3-dev equivalent is libz3-* and ...
-					LATEST_LIBZ3="$(ls -v /usr/lib64/ | grep libz3 | tail -n 1)"
-					sudo ln -sf /usr/lib64/$LATEST_LIBZ3 /usr/lib64/libz3.so.4
+					# sudo zypper install -y $(zypper search libz3 | tail -n 1 | cut -d "|" -f2) libncurses6
 					# toolchain looks for a specific libedit
-					LATEST_LIBEDIT="$(ls -v /usr/lib64/ | grep libedit | tail -n 1)"
-					sudo ln -sf /usr/lib64/$LATEST_LIBEDIT /usr/lib64/libedit.so.2
+					# LATEST_LIBEDIT="$(ls -v /usr/lib64/ | grep libedit | tail -n 1)"
+					# sudo ln -sf /usr/lib64/$LATEST_LIBEDIT /usr/lib64/libedit.so.2
 					;;
 			esac
-			curl -LO https://github.com/CRKatri/llvm-project/releases/download/swift-5.3.2-RELEASE/swift-5.3.2-RELEASE-ubuntu20.04.tar.zst
-			TMP=$(mktemp -d)
-			tar -xvf swift-5.3.2-RELEASE-ubuntu20.04.tar.zst -C $TMP
-			mkdir -p $THEOS/toolchain/linux/iphone $THEOS/toolchain/swift
-			mv $TMP/swift-5.3.2-RELEASE-ubuntu20.04/* $THEOS/toolchain/linux/iphone/
-			ln -s $THEOS/toolchain/linux/iphone $THEOS/toolchain/swift
-			rm -r swift-5.3.2-RELEASE-ubuntu20.04.tar.zst $TMP
+			if [[ $(uname -m) == aarch64 ]]; then
+				curl -sL https://github.com/kabiroberai/swift-toolchain-linux/releases/download/v2.3.0/swift-5.8-ubuntu20.04-aarch64.tar.xz | tar -xJvf - -C $THEOS/toolchain/
+			elif [[ $(uname -m) == x86_64 ]]; then
+				curl -sL https://github.com/kabiroberai/swift-toolchain-linux/releases/download/v2.3.0/swift-5.8-ubuntu20.04.tar.xz | tar -xJvf - -C $THEOS/toolchain/
+			else
+				common "Apologies, we do not currently provide precompiled toolchains for $(uname -m) Linux."
+				get_sdks
+				return
+			fi
 		else
 			case $DISTRO in
 				debian)
-					sudo apt install -y libtinfo5
+					sudo apt install -y libtinfo6
 					;;
 				arch)
-					# get source for libtinfo5 equivalent from AUR and build
-					if ! pacman -Qs ncurses5-compat-libs > /dev/null; then
-						git clone https://aur.archlinux.org/ncurses5-compat-libs.git
-						cd ncurses5-compat-libs
-						gpg --recv-keys "$(cat PKGBUILD | grep validpgp | grep -oP "(?<=').*(?=')")"
-						MAKEFLAGS="-j$(nproc --all)" makepkg -sir --noconfirm \
-							&& cd .. \
-							&& rm -rf ncurses5-compat-libs
-					fi
+					sudo pacman -S --needed --noconfirm ncurses
 					;;
 				redhat)
-					sudo dnf install -y ncurses-compat-libs
+					sudo dnf install -y ncurses-libs
 					;;
 				suse)
-					sudo zypper install -y libncurses5
+					sudo zypper install -y libncurses6
 					;;
 			esac
-			curl -LO https://github.com/L1ghtmann/llvm-project/releases/latest/download/iOSToolchain.tar.xz
-			tar -xvf iOSToolchain.tar.xz -C $THEOS/toolchain/
-			rm iOSToolchain.tar.xz
+			if [[ $(uname -m) == aarch64 || $(uname -m) == x86_64 ]]; then
+				curl -sL https://github.com/L1ghtmann/llvm-project/releases/latest/download/iOSToolchain-$(uname -m).tar.xz | tar -xJvf - -C $THEOS/toolchain/
```

Comment:
it this the right link? both

```
https://github.com/L1ghtmann/llvm-project/releases/latest/download/iOSToolchain-aarch64.tar.xz
https://github.com/L1ghtmann/llvm-project/releases/latest/download/iOSToolchain-x86_64.tar.xz
```

are not found for me

Reply:
never, I went to the repo listed and the most recent release just isn't published yet 👍

Reply:
Yup, want to wait for merge before publishing as the filenames are different and will break the currently revision of the installer which looks for the non-suffixed x64 file.

Reply:
ohh, makes sense. nice